nestkernel/connection_manager.cpp

```diff
@@ -427,7 +426,10 @@ nest::ConnectionManager::connect( const index sgid,
 {
   kernel().model_manager.assert_valid_syn_id( syn_id );
 
-  have_connections_changed_ = true;
+  if ( not have_connections_changed_[ target_thread ] )
```

Comment:
do we need this check here? isn't it cheaper to just write in all cases? the logic wouldn't change, would it?

Reply:
@jakobj I tested on Blaustein, and if we do not check before setting the parameter, then the performance gets a lot worse. You can see it as `4c388f4` [here](https://github.com/nest/nest-simulator/files/3973159/benchmark_results.pdf).  It must have something to do with all the threads trying to access at once and maybe something with the `omp automatic write` in the `set` function. I wrote a comment before every `if`.

Reply:
ok,thanks for clarifying